src/renderer/stores/note.js

```diff
@@ -15,6 +13,44 @@ var Note = Backbone.Model.extend({
       notebookTitle: ''
     }
   },
+
+  initialize: function() {
+    this.dispatchToken = JotzDispatcher.register(this.dispatchCallback.bind(this));
+  },
+
+  dispatchCallback: function(payload) {
+    switch(payload.actionType) {
+      case 'create-block':
+        this.createBlock();
+        break;
+      case 'update-block':
+        this.updateBlock(payload);
+        break;
+      default:
+        break;
+    }
+  },
+
+  createBlock: function() {
+    var blocks = this.get('blocks');
+    blocks.push({
+      language: 'text',
+      content: ''
+    });
+    this.set('blocks', blocks);
+    this.trigger('block-updated');
+  },
+
+  updateBlock: function(payload) {
+    var i = payload.content.index;
+    var blocks = this.get('blocks');
+    blocks[i] = {
+      language: payload.content.language,
+      content: payload.content.content || ''
+    };
+    this.set('blocks', blocks);
```

Comment:
What happens if you close the editor without saving? Will it keep the updated block changes?